jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6134 @@
+	}
+
+	/**
+	 * Swap specified bytes between source address and destination
+	 * 
+	 * @param srcAddress location to start swap
+	 * @param destAddress location to start swap
+	 * @param srcSize the number of bytes to be swapped at source
+	 * @param destSize the number of bytes to be swapped at destination
+	 * 
+	 * @throws IllegalArgumentException if srcAddress or destAddress is illegal, 
+	 *  or if sizes are invalid
+	 */
+	public void copySwapMemory(long srcAddress, long destAddress, long srcSize, long destSize) {
+		copySwapMemory(null, srcAddress, null, destAddress, srcSize, destSize);
+	}
+
+	/**
+	 * Removes the block from the list. Note that the pointers are located 
+	 * immediately before the startIndex value.
+	 * 
+	 * @param startIndex memory address
+	 * 
+	 * @throws IllegalArgumentException if startIndex is not valid
+	 */
+	public void freeMemory(long startIndex) {
+		freeMemoryChecks(startIndex);
+
+		if (0 != startIndex) {
+			freeMemory0(startIndex);
+		}
+	}
+
+	/**
+	 * Returns byte offset to start of class or interface
+	 * 
+	 * @param field which contains desired class or interface
+	 * @return offset to start of class or interface
+	 * 
+	 * @throws NullPointerException if field parameter is null
+	 * @throws IllegalArgumentException if field is static
+	 */
+	public long objectFieldOffset(Field field) {
+		Objects.requireNonNull(field);
+		return objectFieldOffset0(field);
+	}
+
+	/**
+	 * Returns byte offset to start of static class or interface
+	 * 
+	 * @param field which contains desired class or interface
+	 * @return offset to start of class or iterface
+	 * 
+	 * @throws NullPointerException if field parameter is null
+	 * @throws IllegalArgumentException if field is not static
+	 */
+	public long staticFieldOffset(Field field) {
+		Objects.requireNonNull(field);
+		return staticFieldOffset0(field);
+	}
+
+	/**
+	 * Returns static class or interface
+	 * 
+	 * @param field contains desired class or interface
+	 * @return static class or interface in field parameter
+	 * 
+	 * @throws NullPointerException if field parameter is null
+	 * @throws IllegalArgumentException if field is not static
+	 */
+	public Object staticFieldBase(Field field) {
+		Objects.requireNonNull(field);
+		return staticFieldBase0(field);
+	}
+
+	/**
+	 *  Determines whether class has been initialized
+	 * 
+	 * @param class to verify
+	 * @return true if method has not been initialized, false otherwise
+	 * 
+	 * @throws NullPointerException if class is null
+	 */
+	public boolean shouldBeInitialized(Class<?> c) {
+		Objects.requireNonNull(c);
+		return shouldBeInitialized0(c);
+	}
+
+	/** 
+	 * Initializes class parameter if it has not been already
+	 * 
+	 * @param c class to initialize if not already
+	 * 
+	 * @throws NullPointerException if class is null
+	 */
+	public void ensureClassInitialized(Class<?> c) {
+		Objects.requireNonNull(c);
+		ensureClassInitialized0(c);
+	}
+
+	/**
+	 * Return base offset of class array parameter
```

Comment:
Its hard to tell what this function does from the description. It returns the offset in the array object at which the array data storage begins.